docs/pages/api-docs/data-grid/x-grid.md

```diff
@@ -21,6 +21,7 @@ import { XGrid } from '@material-ui/x-grid';
 | <span class="prop-name">autoHeight</span> | <span class="prop-type">boolean</span> | false | If `true`, the grid height is dynamic and follow the number of rows in the grid. |
 | <span class="prop-name">autoPageSize</span> | <span class="prop-type">boolean</span> | false | If `true`, the pageSize is calculated according to the container size and the max number of rows to avoid rendering a vertical scroll bar. |
 | <span class="prop-name">checkboxSelection</span> | <span class="prop-type">boolean</span> | false | If `true`, the grid get a first column with a checkbox that allows to select rows. |
+| <span class="prop-name">checkboxSelectionVisibleOnly</span> | <span class="prop-type">boolean</span> | false | To be used in combination with `checkboxSelection`. If `true`, `Select All` selects only the rows on the current page. |
```

Comment:
Should this option be only xgrid 🤔

Reply:
Can be. It doesn't seem to be that critical to not give it for free. Also since the free version pagination is auto-enabled after a certain amount of rows it makes sense to allow that option as well.

Reply:
Yes, but we could argue that this is for the advanced use cases where users would want to define specific grid behavior.

Reply:
done

Reply:
@oliviertassinari what do you think? Up for putting it in XGrid as it's an advanced configurable grid or prefers to put it in DataGrid as well

Reply:
Regarding DataGrid vs. XGrid, no objections for XGrid only.

---

I can't find this feature in AG Grid. In react-table, I can find it exposed as getToggleAllPageRowsSelectedProps vs. getToggleAllRowsSelectedProps: https://github.com/tannerlinsley/react-table/blob/8ec25cabccf915991ec00cf67bfb26a83aa0c0b2/docs/src/pages/docs/api/useRowSelect.md. react-table. In both cases, the default behavior is to select all the pages. So I think that we are good on this front.

My main question would be: should it be a boolean or an enum? We could imagine something in this order in the long run. 

```jsx
checkboxSelectionMode?: 'all' | 'page' | 'ask'
```

or leave it to the developers, and improve the apiRef to have more than `selectRows()` but something with pre-existing options.

Reply:
I was wondering if there may be a need for more values that's why I called it `checkboxSelectionVisibleOnly`. If in the future we decide to also change the filtering selection that can still fall under the same option because the name is not related to the pagination (the same applies for the infinite loading) 🤔